python/guide/packages/gluon/hybridize.md

```diff
@@ -199,3 +199,81 @@ net(x)
 ```
 
 We can see that the three lines of print statements defined in the `hybrid_forward` function will not print anything. This is because a symbolic program has been produced since the last time `net(x)` was run by calling the `hybridize` function. Afterwards, when we run `net(x)` again, MXNet will no longer need to access Python code, but can directly perform symbolic programming at the C++ backend. This is another reason why model computing performance will be improve after the `hybridize` function is called. However, there is always the potential that any programs we write will suffer a loss in flexibility. If we want to use the three lines of print statements to debug the code in the above example, they will be skipped over and we would not be able to print when the symbolic program is executed. Additionally, in the case of a few functions not supported by Symbol (like `asnumpy`), and operations in-place like `a += b` and `a[:] = a + b` (must be rewritten as `a = a + b`). Therefore, we will not be able to use the `hybrid_forward` function or perform forward computation after the `hybridize` function has been called.
+
+## Key differences and limitations of hybridization
+
+The difference between a purely imperative `Block` and hybridizable `HybridBlock` can superficially appear to be simply the injection of the `F` function space (resolving to [`mx.nd`](http://beta.mxnet.io/api/ndarray/index.html) or [`mx.sym`](http://beta.mxnet.io/api/symbol/index.html)) in the forward function that is renamed from `forward` to `hybrid_forward`. However there are some limitations that apply when using hybrid blocks. In the following section we will review the main differences, giving example of code snippets that generate errors when such blocks get hybridized.
+
+### Indexing
+
+When trying to access specific elements in a tensor like this:
+
+```python
+def hybrid_forward(self, F, x):
+    return x[0,0]
+```
+
+Would generate the following error:
+
+`TypeError: Symbol only support integer index to fetch i-th output`
```

Comment:
Can you elaborate on this error and say that all functions must be supported by both ndarray and symbol and then mention the functions that you have below.